internal/services/netapp/netapp_volume_group_sap_hana_resource.go

```diff
@@ -0,0 +1,628 @@
+			}
+
+			if existing.Model != nil && existing.Model.Id != nil && *existing.Model.Id != "" {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			applicationType := volumegroups.ApplicationTypeSAPNegativeHANA
+
+			volumeList, err := expandNetAppVolumeGroupVolumes(model.Volumes)
+			if err != nil {
+				return err
+			}
+
+			// Performing some basic validations that are not possible in the schema
+			if errorList := validateNetAppVolumeGroupVolumes(volumeList, applicationType); len(errorList) > 0 {
+				return fmt.Errorf("one or more issues found while performing deeper validations for %s:\n%+v", id, errorList)
+			}
+
+			// Parse volume list to set secondary volumes for CRR
+			for i, volumeCrr := range *volumeList {
+				if volumeCrr.Properties.DataProtection != nil &&
+					volumeCrr.Properties.DataProtection.Replication != nil &&
+					strings.EqualFold(string(*volumeCrr.Properties.DataProtection.Replication.EndpointType), string(volumegroups.EndpointTypeDst)) {
+
+					// Modify volumeType as data protection type on main volumeList
+					// so it gets created correctly as data protection volume
+					(*volumeList)[i].Properties.VolumeType = utils.String("DataProtection")
+				}
+			}
+
+			// TODO: deploymentSpecId is temporary until the backend is updated and deploymentSpecId is not required anymore,
+			//       it will be handled internally by the RP
+			deploymentSpecId := "20542149-bfca-5618-1879-9863dc6767f1"
+
+			parameters := volumegroups.VolumeGroupDetails{
+				Location: utils.String(location.Normalize(model.Location)),
+				Properties: &volumegroups.VolumeGroupProperties{
+					GroupMetaData: &volumegroups.VolumeGroupMetaData{
+						GroupDescription:      utils.String(model.GroupDescription),
+						ApplicationType:       &applicationType,
+						ApplicationIdentifier: utils.String(model.ApplicationIdentifier),
+						DeploymentSpecId:      utils.String(deploymentSpecId),
+					},
+					Volumes: volumeList,
+				},
+			}
+
+			err = client.VolumeGroupsCreateThenPoll(ctx, id, parameters)
+			if err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			// Waiting for volume group be completely provisioned
+			if err := waitForVolumeGroupCreateOrUpdate(ctx, client, id); err != nil {
+				return err
+			}
+
+			// CRR - Authorizing secondaries from primary volumes
+			for _, volumeCrr := range *volumeList {
+				if volumeCrr.Properties.DataProtection != nil &&
+					volumeCrr.Properties.DataProtection.Replication != nil &&
+					strings.EqualFold(string(*volumeCrr.Properties.DataProtection.Replication.EndpointType), string(volumegroups.EndpointTypeDst)) {
+
+					// Getting secondary volume resource id
+					secondaryId := volumes.NewVolumeID(subscriptionId,
+						model.ResourceGroupName,
+						model.AccountName,
+						getResourceNameString(volumeCrr.Properties.CapacityPoolResourceId),
+						getResourceNameString(volumeCrr.Name),
```

Comment:
I don't think we need this either as the name is just name or can this sometimes be something else?

Reply:
You're right with that one, I'll fix this.